pandas/core/frame.py

```diff
@@ -1964,6 +1965,29 @@ def to_feather(self, path):
 
         to_feather(self, path)
 
+    def to_markdown(self):
+        """
+        Print a DataFrame in markdown-friendly format.
+
+        .. versionadded:: 1.0
+
+        Returns
+        -------
+        str
+            DataFrame in markdown-friendly format.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame(data={'col1': [1, 2], 'col2': [3, 4]})
+        >>> print(df.to_markdown())
+        |    |   col1 |   col2 |
+        |---:|-------:|-------:|
+        |  0 |      1 |      3 |
+        |  1 |      2 |      4 |
+        """
+        tabulate = import_optional_dependency("tabulate")
+        return self.pipe(tabulate.tabulate, headers="keys", tablefmt="pipe")
```

Comment:
id have to go look up how `pipe` works.  is there a lower-tech way of doing this?

Reply:
`tabulate.tabulate(self, headers="keys", tablefmt="pipe")` I suspect.